internal/quickfort/place.lua

```diff
@@ -45,23 +46,23 @@ local function is_valid_stockpile_extent(s)
 end
 
 local stockpile_db = {
-    a={label='Animal'},
-    f={label='Food'},
-    u={label='Furniture'},
-    n={label='Coins'},
-    y={label='Corpses'},
-    r={label='Refuse'},
-    s={label='Stone'},
-    w={label='Wood'},
-    e={label='Gem'},
-    b={label='Bar/Block'},
-    h={label='Cloth'},
-    l={label='Leather'},
-    z={label='Ammo'},
-    S={label='Sheets'},
-    g={label='Finished Goods'},
-    p={label='Weapons'},
-    d={label='Armor'},
+    a={label='Animal', index=0},
```

Comment:
You could theoretically set `viewscreen_layer_stockpilest.layer_objects[0].cursor` to `index` directly, instead sending the down arrow repeatedly. One downside is that there's no guarantee that Toady won't move objects within `layer_objects` in the future, and there's little type-safety if that happens (it would only fail immediately if entry 0 changes from a `layer_object_listst` to a `layer_object_buttonst`).

Another even more fun approach would be to read tiles from the screen with `readTile()`, although that's probably way more brittle than your approach here).

Honestly, the stockpile screen is pretty stable, so your approach is fine. These are just a couple others that might be useful for other things.

Reply:
interestingly, a screen scraping version of "readTile" is how python quickfort implemented some of its fancier features, like material autoselection. I think we'll have better options for our quickfort, though, since we have access to internal data structures.

If the UI changes or the lists change contents/order, a whole lot of query blueprints are going to break : p  I've tried to centralize a lot of the query logic in the common aliases file (next PR) so many of the fixes can be "soft" -- that is, we can fix blueprints by just updating the aliases that come with dfhack.

I'm watching what Toady is doing for the Steam UI. His attempts to make the UI cleaner might require a bunch of changes to this script and its data files.

I've been testing some heavy query blueprints, and so far I'm pretty satisfied with their performance -- I can toggle all the hundreds of tallow entries in a food stockpile in a fraction of a second, for example -- so I've been pretty happy with the simple approach of just sending keystrokes.

Reply:
That's a good point that I forgot about - any change to the screen would also break the arrow keys approach. It's hard to detect these sorts of things automatically, but hopefully we'll catch any issues that arise in future releases.